src/OrchardCore.Modules/OrchardCore.Liquid/Filters/ContainersFilter.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.ContentManagement;
+
+namespace OrchardCore.Liquid.Filters
+{
+    public class ContainersFilter : ILiquidFilter
+    {
+        private readonly IContentManager _contentManager;
+
+        public ContainersFilter(IContentManager contentManager)
+        {
+            _contentManager = contentManager;
+        }
+
+        public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
+        {
+            var contentItem = input.ToObjectValue() as ContentItem;
+
+            if (contentItem == null)
+            {
+                throw new ArgumentException("A Content Item was expected");
+            }
+
+            var containers = new List<ObjectValue>();
+
+            string containerId = contentItem.Content?.ContainedPart?.ListContentItemId;
+
+            while (containerId != null)
+            {
+                contentItem = await _contentManager.GetAsync(containerId);
```

Comment:
@arkadiuszwojcik 

I suggest to define your liquid filter in the `OC.Lists` module, there are already other filters in this module under a `Liquid` folder, and then register it in the startup of the same module, this through the `LiquidStartup` that will be executed only if the `Lists` feature is enabled and also only if the `Liquid` feature is enabled, because of the related `[RequireFeatures("OrchardCore.Liquid")]` attribute.

    [RequireFeatures("OrchardCore.Liquid")]
    public class LiquidStartup : StartupBase
    {
        public override void ConfigureServices(IServiceCollection services)
        {
            services.AddLiquidFilter<ListCountFilter>("list_count");
            services.AddLiquidFilter<ListItemsFilter>("list_items");
            // Register here your liquid filter
        }
    }

Reply:
Good idea. In such case "list_ancestors" or "list_item_ancestors" will be better filter name.